File: compile_less.js
#!/usr/bin/env node

const fs = require('fs');
const path = require('path');
const less = require('less');

const baseDir = 'c:\\Users\\Administrator\\.workbuddy\\FreeCDN\\web\\views\\@default';

const filesToCompile = [
    { input: '@design_tokens.less', output: '@design_tokens.css' },
    { input: '@components_base.less', output: '@components_base.css' },
    { input: '@components_data.less', output: '@components_data.css' },
    { input: '@components_feedback.less', output: '@components_feedback.css' },
    { input: '@globals_utilities.less', output: '@globals_utilities.css' },
    { input: 'dashboard/index.less', output: 'dashboard/index.css' },
    { input: 'clusters/index.less', output: 'clusters/index.css' },
    { input: 'servers/index.less', output: 'servers/index.css' },
    { input: 'users/features.less', output: 'users/features.css' },
    { input: 'nodes/index.less', output: 'nodes/index.css' },
    { input: 'admins/index.less', output: 'admins/index.css' },
    { input: 'settings/index.less', output: 'settings/index.css' },
    { input: 'dns/index.less', output: 'dns/index.css' },
    { input: 'index/index.less', output: 'index/index.css' },
    { input: 'log/index.less', output: 'log/index.css' },
    { input: 'setup/index.less', output: 'setup/index.css' },
    { input: 'db/index.less', output: 'db/index.css' },
    { input: 'ui/index.less', output: 'ui/index.css' },
];


console.log('='.repeat(60));
console.log('LESS 编译脚本（Node.js）');
console.log('='.repeat(60));

async function compile(inputFile, outputFile) {
    const inputPath = path.join(baseDir, inputFile);
    const outputPath = path.join(baseDir, outputFile);
    
    try {
        const lessCode = fs.readFileSync(inputPath, 'utf8');
        
        console.log(`\n编译 ${inputFile}...`);
        
        const result = await less.render(lessCode, {
            filename: inputPath,
            paths: [baseDir]
        });
        
        fs.writeFileSync(outputPath, result.css, 'utf8');
        console.log(`✓ 编译成功: ${outputFile}`);
        return true;
    } catch (error) {
        console.log(`✗ 编译失败: ${inputFile}`);
        console.log(`  错误: ${error.message}`);
        return false;
    }
}

async function compileAll() {
    let success = 0;
    let fail = 0;
    
    for (const file of filesToCompile) {
        const result = await compile(file.input, file.output);
        if (result) success++;
        else fail++;
    }
    
    console.log('\n' + '='.repeat(60));
    console.log(`编译完成: ${success} 成功, ${fail} 失败`);
    console.log('='.repeat(60));
}

compileAll().catch(console.error);

File: compile_less.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import subprocess
import os
import sys

os.chdir(r"c:\Users\Administrator\.workbuddy\FreeCDN\web\views\@default")

files_to_compile = [
    ("@design_tokens.less", "@design_tokens.css"),
    ("@components_base.less", "@components_base.css"),
    ("@components_data.less", "@components_data.css"),
    ("@components_feedback.less", "@components_feedback.css"),
    ("@globals_utilities.less", "@globals_utilities.css"),
    ("dashboard/index.less", "dashboard/index.css"),
    ("clusters/index.less", "clusters/index.css"),
    ("servers/index.less", "servers/index.css"),
    ("users/features.less", "users/features.css"),
]

print("=" * 60)
print("LESS 编译脚本")
print("=" * 60)

for less_file, css_file in files_to_compile:
    try:
        print(f"\n编译 {less_file}...")
        result = subprocess.run(
            ["lessc", less_file, css_file],
            capture_output=True,
            text=True,
            timeout=10
        )
        
        if result.returncode == 0:
            print(f"✓ 编译成功: {css_file}")
        else:
            print(f"✗ 编译失败: {less_file}")
            if result.stderr:
                print(f"  错误: {result.stderr[:200]}")
            if result.stdout:
                print(f"  输出: {result.stdout[:200]}")
    except Exception as e:
        print(f"✗ 错误: {str(e)}")

print("\n" + "=" * 60)
print("编译完成")
print("=" * 60)
